### ISSUE TITLE
containerd-recombine log operator errors on 0.49.0

### ISSUE THREAD

**Issue by KingJ** [signalfx/splunk-otel-collector-chart#443]
I'm experimenting with chart & collector v0.49.0, however in clusters that use containerd (e.g. GKE based clusters), i'm seeing an unexpected error related to log collection in the OTEL container logs;

```
	Failed to process entry	{"kind": "receiver", "name": "filelog", "operator_id": "containerd-recombine", "operator_type": "recombine", "error": "invalid operation: int(string) (1:7)\n | (body.logtag) == 'F'\n | ......^", "action": "send", "entry": {"observed_timestamp":"2022-04-29T13:23:45.433690115Z","timestamp":"2022-04-29T13:23:45.388206949Z","body":"2022-04-29T13:23:45.388206949Z stdout F 2022-04-29 13:23:45.387 [INFO][59] monitor-addresses/startup.go 714: Using autodetected IPv4 address on interface eth0: 172.19.72.7/32","attributes":{"log":"2022-04-29 13:23:45.387 [INFO][59] monitor-addresses/startup.go 714: Using autodetected IPv4 address on interface eth0: 172.19.72.7/32","log.file.path":"/var/log/pods/kube-system_calico-node-7rsmm_23d2706b-4b34-4b0b-addd-936f90a6b757/calico-node/0.log","logtag":"F","stream":"stdout"},"severity":0,"scope_name":""}}
github.com/open-telemetry/opentelemetry-log-collection/operator/helper.(*TransformerOperator).HandleEntryError
	/builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/github.com/open-telemetry/opentelemetry-log-collection@v0.29.0/operator/helper/transformer.go:110
github.com/open-telemetry/opentelemetry-log-collection/operator/transformer/recombine.(*RecombineOperator).Process
	/builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/github.com/open-telemetry/opentelemetry-log-collection@v0.29.0/operator/transformer/recombine/recombine.go:207
github.com/open-telemetry/opentelemetry-log-collection/operator/helper.(*WriterOperator).Write
	/builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/github.com/open-telemetry/opentelemetry-log-collection@v0.29.0/operator/helper/writer.go:65
github.com/open-telemetry/opentelemetry-log-collection/operator/helper.(*ParserOperator).ProcessWithCallback
	/builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/github.com/open-telemetry/opentelemetry-log-collection@v0.29.0/operator/helper/parser.go:128
github.com/open-telemetry/opentelemetry-log-collection/operator/helper.(*ParserOperator).ProcessWith
	/builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/github.com/open-telemetry/opentelemetry-log-collection@v0.29.0/operator/helper/parser.go:104
github.com/open-telemetry/opentelemetry-log-collection/operator/parser/regex.(*RegexParser).Process
	/builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/github.com/open-telemetry/opentelemetry-log-collection@v0.29.0/operator/parser/regex/regex.go:103
github.com/open-telemetry/opentelemetry-log-collection/operator/transformer/router.(*RouterOperator).Process
	/builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/github.com/open-telemetry/opentelemetry-log-collection@v0.29.0/operator/transformer/router/router.go:135
github.com/open-telemetry/opentelemetry-log-collection/operator/helper.(*WriterOperator).Write
	/builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/github.com/open-telemetry/opentelemetry-log-collection@v0.29.0/operator/helper/writer.go:65
github.com/open-telemetry/opentelemetry-log-collection/operator/input/file.(*Reader).emit
	/builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/github.com/open-telemetry/opentelemetry-log-collection@v0.29.0/operator/input/file/reader.go:198
github.com/open-telemetry/opentelemetry-log-collection/operator/input/file.(*Reader).ReadToEnd
	/builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/github.com/open-telemetry/opentelemetry-log-collection@v0.29.0/operator/input/file/reader.go:142
github.com/open-telemetry/opentelemetry-log-collection/operator/input/file.(*InputOperator).poll.func1
	/builds/o11y-gdi/splunk-otel-collector-releaser/.go/pkg/mod/github.com/open-telemetry/opentelemetry-log-collection@v0.29.0/operator/input/file/file.go:155
2022-04-29T13:23:55.633Z	error	helper/transformer.go:110
```

This error repeats frequently for a range of different log files. 

If I roll back to 0.48.0, I do not see this error and log ingestion works as normal. This error is also not present on clusters which are still using Docker as their container runtime.

**Comment by jvoravong** [comment]
Was able to reproduce this issue, we'll work on getting a fix out.

**Comment by dmitryax** [comment]
@KingJ the issue should be fixed in 0.50.0. Please let us know how it works

**Comment by KingJ** [comment]
@dmitryax I've given 0.50.0 a try and I can confirm it is now working on our containerd-based clusters! Many thanks!
